libs/src/api/serviceProvider.js

```diff
@@ -23,15 +23,29 @@ class ServiceProvider extends Base {
    * Fetches healthy creator nodes and autoselects a primary
    * and two secondaries
    * @param {number} numberOfNodes total number of nodes to fetch (2 secondaries means 3 total)
-   * @param {Set<string} blacklist whether or not to exclude any nodes
+   * @param {Set<string>} whitelist whether or not to include only specified nodes (default no whiltelist)
+   * @param {Set<string} blacklist whether or not to exclude any nodes (default no blacklist)
    * @returns { primary, secondaries, services }
    * // primary: string
    * // secondaries: Array<string>
    * // services: { creatorNodeEndpoint: versionInfo }
    */
-  async autoSelectCreatorNodes (numberOfNodes = 3, blacklist = new Set([])) {
+  async autoSelectCreatorNodes (
+    numberOfNodes = 3,
+    whitelist = null,
+    blacklist = new Set([])
```

Comment:
any reason you don't init whitelist to empty set as well?

Reply:
because a whitelist of empty set would mean "filter out everything"

Reply:
ah yes i dumb

Reply:
changed blacklist to be null default!